From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@niif.hu>
Date: Tue, 22 Dec 2015 22:22:49 +0100
Subject: Clean up generated files

Removed the clean-generic rules, because they override CLEANFILES,
and mostly contain irrelevant generic stuff, with the exception of
pengine, which is handled specially.
---
 crmd/Makefile.am    |    5 ++---
 doc/Makefile.am     |    7 ++++---
 fencing/Makefile.am |    2 ++
 lrmd/Makefile.am    |    2 ++
 mcp/Makefile.am     |    4 ++--
 pengine/Makefile.am |    4 ++--
 tools/Makefile.am   |    5 ++---
 xml/Makefile.am     |    3 +--
 8 files changed, 17 insertions(+), 15 deletions(-)

diff --git a/crmd/Makefile.am b/crmd/Makefile.am
index 984f5d0..c28da0b 100644
--- a/crmd/Makefile.am
+++ b/crmd/Makefile.am
@@ -52,9 +52,6 @@ if BUILD_XML_HELP
 man7_MANS =	crmd.7
 endif
 
-clean-generic:
-	rm -f *.log *.debug *.xml *~
-
 graphs:	fsa_inputs.png fsa_inputs_by_action.png fsa_actions_by_state.png
 
 %.png: %.dot
@@ -62,3 +59,5 @@ graphs:	fsa_inputs.png fsa_inputs_by_action.png fsa_actions_by_state.png
 
 %.dot : fsa_matrix.h make_dot.pl
 	perl $(top_srcdir)/crmd/make_dot.pl $(top_srcdir)/crmd/fsa_matrix.h $(top_builddir)/crmd
+
+CLEANFILES = $(man7_MANS)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 3077276..f4c836d 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -38,9 +38,10 @@ DOCBOOK_LANGS   := en-US
 DOTs = $(wildcard */en-US/images/*.dot)
 SVG = $(wildcard */en-US/images/pcmk-*.svg) $(DOTs:%.dot=%.svg)
 
-PNGS = $(SVG:%.svg=%-small.png) $(SVG:%.svg=%.png)  $(SVG:%.svg=%-large.png) 	\
+generated_PNGS = $(SVG:%.svg=%-small.png) $(SVG:%.svg=%.png)  $(SVG:%.svg=%-large.png) 	\
 	Pacemaker_Explained/en-US/images/Policy-Engine-big.png  \
-	Pacemaker_Explained/en-US/images/Policy-Engine-small.png \
+	Pacemaker_Explained/en-US/images/Policy-Engine-small.png
+PNGS = $(generated_PNGS) \
 	Pacemaker_Remote/en-US/images/pcmk-ha-cluster-stack.png \
 	Pacemaker_Remote/en-US/images/pcmk-ha-remote-stack.png
 
@@ -200,6 +201,6 @@ if BUILD_DOCBOOK
 endif
 
 clean-local:
-	-rm -rf $(generated_docs) $(generated_mans) $(docbook_build) ../xsl
+	-rm -rf $(generated_docs) $(generated_mans) $(docbook_build) ../xsl $(generated_PNGS)
 	-rm -rf $(SHARED_XML) $(CFS_XML) $(PE_XML) $(PR_XML)
 	for book in $(docbook); do rm -rf $$book/tmp $$book/publish; done
diff --git a/fencing/Makefile.am b/fencing/Makefile.am
index f422e82..383c217 100644
--- a/fencing/Makefile.am
+++ b/fencing/Makefile.am
@@ -72,3 +72,5 @@ stonithd_LDADD		= $(top_builddir)/lib/common/libcrmcommon.la	\
 			$(CRYPTOLIB) $(CLUSTERLIBS)
 
 CFLAGS			= $(CFLAGS_COPY:-Werror=)
+
+CLEANFILES = $(man7_MANS) $(man8_MANS)
diff --git a/lrmd/Makefile.am b/lrmd/Makefile.am
index 6d061de..a8fb07a 100644
--- a/lrmd/Makefile.am
+++ b/lrmd/Makefile.am
@@ -61,3 +61,5 @@ lrmd_test_LDADD		= $(top_builddir)/lib/common/libcrmcommon.la	\
 			$(top_builddir)/pengine/libpengine.la
 
 noinst_HEADERS  = lrmd_private.h
+
+CLEANFILES = $(man8_MANS)
diff --git a/mcp/Makefile.am b/mcp/Makefile.am
index c3caf19..1b3720a 100644
--- a/mcp/Makefile.am
+++ b/mcp/Makefile.am
@@ -35,6 +35,6 @@ pacemakerd_SOURCES	= pacemaker.c corosync.c
 pacemakerd_LDADD	= $(top_builddir)/lib/cluster/libcrmcluster.la $(top_builddir)/lib/common/libcrmcommon.la
 pacemakerd_LDADD	+= $(CLUSTERLIBS)
 
-clean-generic:
-	rm -f *.log *.debug *.xml *~
 endif
+
+CLEANFILES = $(man8_MANS)
diff --git a/pengine/Makefile.am b/pengine/Makefile.am
index d44178f..95d639e 100644
--- a/pengine/Makefile.am
+++ b/pengine/Makefile.am
@@ -82,5 +82,5 @@ install-exec-local:
 
 uninstall-local:
 
-clean-generic:
-	rm -f test10/*.pe.*
+clean-local:
+	rm -f test10/*.pe.* $(man7_MANS)
diff --git a/tools/Makefile.am b/tools/Makefile.am
index b43130e..555b1db 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -143,11 +143,10 @@ ipmiservicelogd_CFLAGS	= $(OPENIPMI_SERVICELOG_CFLAGS) $(SERVICELOG_CFLAGS)
 ipmiservicelogd_LDFLAGS	= $(top_builddir)/lib/common/libcrmcommon.la $(OPENIPMI_SERVICELOG_LIBS) $(SERVICELOG_LIBS)
 endif
 
-clean-generic:
-	rm -f *.log *.debug *.xml *~
-
 install-exec-local:
 
 uninstall-local:
 
 .PHONY: install-exec-hook
+
+CLEANFILES = $(man8_MANS)
diff --git a/xml/Makefile.am b/xml/Makefile.am
index 81ba763..5037089 100644
--- a/xml/Makefile.am
+++ b/xml/Makefile.am
@@ -125,5 +125,4 @@ sync:
 	git rm -f $(wildcard *-next.rng)
 	make pacemaker-next.rng
 
-clean:
-	rm -f $(RNG_generated)
+CLEANFILES = $(RNG_generated)
